src/dirs.proj

```diff
@@ -15,6 +15,10 @@
     <Project Include="shims\ApiCompat.proj" />
   </ItemGroup>
 
+  <ItemGroup Condition="'$(DirectoryToBuild)' != ''">
+    <Project Include="$(DirectoryToBuild)/**/*.csproj" />
```

Comment:
I didn't really want to add the cost of vbproj just for the one project. Similar for vcxproj, I was thinking about adding some custom logic that looks for a dirs.proj in the directory and uses that instead if it exists and that will allow directories to override the specific logic for building that library.